src/stores/attachment_store.js

```diff
@@ -29,11 +29,11 @@ class AttachmentsStore extends Store {
     });
   }
 
-  getAttachment(commentId) {
-    var attachment = assign({}, attachments.get(commentId));
+  getAttachments(commentId) {
+    var a = assign({}, attachments.get(commentId));
     attachments = attachments.delete(commentId);
```

Comment:
I'm still reading through the code to understand it all, so this concern might not be valid, but altering a store's state in a getter seems strange to me.